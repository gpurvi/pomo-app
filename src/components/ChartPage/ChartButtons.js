import React from 'react';
// import ChartDatePicker from "./ChartDatePicker";
import DatePickerV1 from "../common/DatePickerV1";

const ChartButtons = (props) => (
    <div>
        <label>
            <input
                type="checkbox"
                value="sessions"
                checked={props.showSessions}
                onChange={props.onCheck}
            />
            Sessions
        </label>
        <label>
            <input
                type="checkbox"
                value="hours"
                checked={props.showDurations}
                onChange={props.onCheck}
            />
            Hours
        </label>
        <DatePickerV1
            minDate={props.minDate}
            maxDate={props.maxDate}
            onDateChange={props.onDateChange}
            date={props.date}
            today={false}
            maxDetail={props.maxDetail}
        />
        {/*<ChartDatePicker*/}
        {/*nextDisabled={props.nextDisabled}*/}
        {/*prevDisabled={props.prevDisabled}*/}
        {/*minDate={props.minDate}*/}
        {/*maxDate={props.maxDate}*/}
        {/*maxDetail={props.maxDetail}*/}
        {/*onDateChange={props.onDateChange}*/}
        {/*date={props.date}*/}
        {/*onDateChangeClick={props.onDateChangeClick}*/}
        {/*/>*/}
    </div>
);

export default ChartButtons;