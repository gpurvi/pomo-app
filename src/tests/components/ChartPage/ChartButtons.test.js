import React from 'react';
import {shallow} from 'enzyme';
import ChartButtons from '../../../components/ChartPage/ChartButtons';

it('should render ChartButtons', () => {
    const wrapper = shallow(<ChartButtons/>);
    expect(wrapper).toMatchSnapshot();
});

it('should call props.onCheck on sessions input', () => {
    const onCheck = jest.fn();
    const wrapper = shallow(<ChartButtons onCheck={onCheck}/>);
    wrapper.find('input').first().simulate('change');
    expect(onCheck).toHaveBeenCalled();
});

it('should call props.onCheck on hours input', () => {
    const onCheck = jest.fn();
    const wrapper = shallow(<ChartButtons onCheck={onCheck}/>);
    wrapper.find('input').last().simulate('change');
    expect(onCheck).toHaveBeenCalled();
});