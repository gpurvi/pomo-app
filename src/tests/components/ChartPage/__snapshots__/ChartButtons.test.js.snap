// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChartButtons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartButtons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          <input
                    checked={undefined}
                    onChange={undefined}
                    type="checkbox"
                    value="sessions"
          />
          Sessions
</label>,
        <label>
          <input
                    checked={undefined}
                    onChange={undefined}
                    type="checkbox"
                    value="hours"
          />
          Hours
</label>,
        <DatePickerV1
          date={undefined}
          maxDetail={undefined}
          minDate={undefined}
          onDateChange={undefined}
          today={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={undefined}
              onChange={undefined}
              type="checkbox"
              value="sessions"
/>,
            "Sessions",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": undefined,
              "onChange": undefined,
              "type": "checkbox",
              "value": "sessions",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Sessions",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={undefined}
              onChange={undefined}
              type="checkbox"
              value="hours"
/>,
            "Hours",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": undefined,
              "onChange": undefined,
              "type": "checkbox",
              "value": "hours",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Hours",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "date": undefined,
          "maxDetail": undefined,
          "minDate": undefined,
          "onDateChange": undefined,
          "today": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            <input
                        checked={undefined}
                        onChange={undefined}
                        type="checkbox"
                        value="sessions"
            />
            Sessions
</label>,
          <label>
            <input
                        checked={undefined}
                        onChange={undefined}
                        type="checkbox"
                        value="hours"
            />
            Hours
</label>,
          <DatePickerV1
            date={undefined}
            maxDetail={undefined}
            minDate={undefined}
            onDateChange={undefined}
            today={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={undefined}
                onChange={undefined}
                type="checkbox"
                value="sessions"
/>,
              "Sessions",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": undefined,
                "onChange": undefined,
                "type": "checkbox",
                "value": "sessions",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Sessions",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={undefined}
                onChange={undefined}
                type="checkbox"
                value="hours"
/>,
              "Hours",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": undefined,
                "onChange": undefined,
                "type": "checkbox",
                "value": "hours",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Hours",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "date": undefined,
            "maxDetail": undefined,
            "minDate": undefined,
            "onDateChange": undefined,
            "today": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
